### DISCUSSION TITLE
Add possibility to show the description of a formula with `brew list` and `brew leaves`

### DISCUSSION THREAD

**Discussion by nbro** [Homebrew/discussions#5344] [Writing Formulae/Casks]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.2.20
ORIGIN: https://github.com/Homebrew/brew
HEAD: c2ed3327c605c3e738359c9807b8f4cd6fec09eb
Last commit: 8 days ago
Core tap HEAD: d6500a8912b02f838d9edf36263f6487fb8af34d
Core tap last commit: 26 hours ago
Core tap JSON: 06 May 12:34 UTC
Core cask tap HEAD: 8f1cdf8d25c0f626b622362cedaf76c4c428c6f5
Core cask tap last commit: 26 hours ago
Core cask tap JSON: 06 May 12:34 UTC
HOMEBREW_PREFIX: /usr/local
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 8
Homebrew Ruby: 3.1.4 => /usr/local/Homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: octa-core 64-bit haswell
Clang: 13.1.6 build 1316
Git: 2.32.1 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 12.7.4-x86_64
CLT: 14.2.0.0.1.1668646533
Xcode: 13.4.1
```

### Output of `brew doctor`

```shell
﻿﻿﻿﻿Please note that these warnings are just used to help the Homebrew maintainers
with debugging if you file an issue. If everything you use Homebrew for is
working fine: please don't worry or file an issue; just ignore this. Thanks!

Warning: You have the following deprecated, cask taps tapped:
  caskroom/cask
Untap them with `brew untap`.

Warning: Some installed formulae are deprecated or disabled.
You should find replacements for the following formulae:
  openssl@1.1
  python@3.8

Warning: You have unlinked kegs in your Cellar.
Leaving kegs unlinked can lead to build-trouble and cause formulae that depend on
those kegs to fail to run properly once built. Run `brew link` on these:
  glib
  numpy
  kubernetes-cli
  gnupg

Warning: You have an unnecessary local Cask tap.
This can cause problems installing up-to-date casks.
Please remove it by running:
  brew untap homebrew/cask

Warning: You have an unnecessary local Core tap!
This can cause problems installing up-to-date formulae.
Please remove it by running:
 brew untap homebrew/core

Warning: Homebrew's "sbin" was not found in your PATH but you have installed
formulae that put executables in /usr/local/sbin.
Consider setting your PATH for example like so:
  echo 'export PATH="/usr/local/sbin:$PATH"' >> ~/.zshrc

Warning: Your Xcode (13.4.1) is outdated.
Please update to Xcode 14.2 (or delete it).
Xcode can be updated from the App Store.
```

### Description of issue

It would be nice to be able to see the description of each formula whenever we do `brew list`, so that we have an idea of what is installed in our system, if we don't know a formula (because maybe it was installed as a dependency of another one or you simply don't know or remember why you installed it). So, basically, what I'm asking is that `brew list` gets another flag `--desc`, which can be used with `--versions` or `--installed-on-request` (or any other flag) and that would show the description of a formula. I know there's `brew desc`, but that is not the same thing. 

See also https://github.com/orgs/Homebrew/discussions/5342

**Comment by carlocab** [comment]
Have you tried doing something like
```sh
brew list | xargs brew desc
brew desc $(brew list)
brew leaves | xargs brew desc
brew desc $(brew leaves)
```
?

**Reply by nbro** [reply]
Yes, I thought of using `xargs` too, but what if I want to see the description and version at the same time. In other words, I'd like to be able to combine different flags when calling `list`. I don't think this might be that difficult to implement. If I can help, point me to the right code that would need to be changed.

**Reply by carlocab** [reply]
> what if I want to see the description and version at the same time

You can use something like
```sh
brew leaves | xargs -I{} sh -c 'brew info {} | head -2'
```
or even
```
brew info --json=v2 --installed --formulae | jq -r '.formulae.[] | (.name + " (" + .versions.stable + "): " + .desc)'
```

> I don't think this might be that difficult to implement.

I agree that it would not be difficult to implement -- I'm just not sure that it's something enough users are interested in enough for it to be worth maintaining code for. The users who are interested can use `brew info --json` to extract whatever information they need from `brew` in a format that's relatively straightforward to manipulate.

However, other maintainers may disagree, so:

> If I can help, point me to the right code that would need to be changed.

You are probably interested in the code at

https://github.com/Homebrew/brew/blob/020d9947b0a195c9f91b264a0ebf8dc87d6b14d3/Library/Homebrew/cmd/list.rb
https://github.com/Homebrew/brew/blob/020d9947b0a195c9f91b264a0ebf8dc87d6b14d3/Library/Homebrew/cmd/leaves.rb
